dataactvalidator/validation_handlers/validator.py

```diff
@@ -63,58 +63,66 @@ def validate(cls, record, csvSchema):
-            checkRequiredOnly = False
-            currentSchema = csvSchema[fieldName]
+            check_required_only = False
+            current_schema = csv_schema[field_name]
+            total_fields += 1
 
-            currentData = record[fieldName]
-            if(currentData != None):
-                currentData = currentData.strip()
+            current_data = record[field_name]
+            if current_data is not None:
+                current_data = current_data.strip()
 
-            if(currentData == None or len(currentData) == 0):
-                if(currentSchema.required ):
+            if current_data is None or len(current_data) == 0:
+                blank_fields += 1
+                if current_schema.required:
                     # If empty and required return field name and error
-                    recordFailed = True
-                    failedRules.append([fieldName, ValidationError.requiredError, "", "", "fatal"])
+                    record_failed = True
+                    failed_rules.append([field_name, ValidationError.requiredError, "", "", "fatal"])
                     continue
                 else:
                     # If field is empty and not required its valid
-                    checkRequiredOnly = True
+                    check_required_only = True
 
             # Always check the type in the schema
-            if(not checkRequiredOnly and not Validator.checkType(currentData,currentSchema.field_type.name) ) :
-                recordTypeFailure = True
-                recordFailed = True
-                failedRules.append([fieldName, ValidationError.typeError, currentData,"", "fatal"])
+            if not check_required_only and not Validator.checkType(current_data,current_schema.field_type.name):
+                record_type_failure = True
+                record_failed = True
+                failed_rules.append([field_name, ValidationError.typeError, current_data,"", "fatal"])
                 # Don't check value rules if type failed
                 continue
 
             # Check length based on schema
-            if currentSchema.length is not None and currentData is not None and len(currentData.strip()) > currentSchema.length:
+            if current_schema.length is not None and current_data is not None and len(current_data.strip()) > current_schema.length:
                 # Length failure, add to failedRules
-                recordFailed = True
-                failedRules.append([fieldName, ValidationError.lengthError, currentData,"", "warning"])
-
-        return (not recordFailed), failedRules, (not recordTypeFailure)
+                record_failed = True
+                failed_rules.append([field_name, ValidationError.lengthError, current_data,"", "warning"])
+
+        # if all columns are blank (empty row), set it so it doesn't add to the error messages or write the line, just ignore it
+        if total_fields == blank_fields:
+            record_failed = False
```

Comment:
Not a bad risk. If the `blank_fields` total is equal to the `total_fields` then everything in there is blank and it shouldn't matter. It will actually override it because, technically, it's going to set it to `True` earlier because we're missing required column headers, but we want it to be `False` so it can be ignored. It's only checking one row at a time so we won't be messing up `record_failed` for other rows.